docs/datamodel/links.rst

```diff
@@ -14,16 +14,265 @@ There are two kinds of link item declarations: *abstract links*, and
 *concrete links*.  Abstract links are defined on the module level and are
 not tied to any particular object type. Typically this is done to set
 some :ref:`annotations <ref_datamodel_annotations>`, define
-:ref:`link properties <ref_datamodel_props>`, or setup :ref:`constraints
-<ref_datamodel_constraints>`.  Concrete links are defined on specific object
-types.
+:ref:`link properties <ref_datamodel_props>`, or setup
+:ref:`constraints <ref_datamodel_constraints>`.  Concrete links
+are defined on specific object types.
 
 Links are directional and have a *source* and one or more *targets*.
 The number of targets is specified by the keywords :ref:`required
 <ref_eql_ddl_links_syntax>`, :ref:`single <ref_eql_ddl_links_syntax>`,
 and :ref:`multi <ref_eql_ddl_links_syntax>`.  It is also possible to
 restrict how many source objects can link to the same target via the
-:eql:constraint:`exclusive` constraint.
+:eql:constraint:`exclusive` constraint. Using these tools it's
+possible to specify common relationships between things:
+*one-to-one*, *one-to-many*, and *many-to-many*.
+
+
+One-to-One
+----------
+
+A *one-to-one* relationship represents a situation where everything
+is broken up into pairs and all the pairs are unique. For example,
+``Employee`` and ``ReservedParking``:
+
+.. code-block:: sdl
+
+    type Employee {
+        required property name -> str;
+        single link parking -> ReservedParking {
+            constraint exclusive;
+        }
+    }
+    type ReservedParking {
+        required property number -> int64;
+    }
+
+An ``Employee`` can have up to one ``ReservedParking`` assigned
+exclusively to them. The :eql:constraint:`exclusive` constraint
+ensures that no more than *one* ``Employee`` can get the same
+``ReservedParking``, while the ``single`` qualifier on the link (which
+is the default, so it can be omitted) ensures that no ``Employee`` can
+have more than *one* ``ReservedParking``. Together the constraint and
+the qualifier specify a *one-to-one* relationship.
+
+Although the link is specified only on one of the objects, the
+relationship involves both of them and so it can be accessed from
+either end. To get the assigned ``ReservedParking`` given an
+``Employee`` the following query can be used:
+
+.. code-block:: edgeql
+
+    WITH Alice := (
+        SELECT Employee FILTER .name = 'Alice'
+    )
+    SELECT Alice.parking {
+        number
+    };
+
+The reverse lookup of who owns a particular ``ReservedParking`` spot
+can be done by using a backward link traversal like so:
+
+.. code-block:: edgeql
+
+    WITH Spot := (
+        SELECT ReservedParking FILTER .number = 42
+    )
+    SELECT Spot.<parking[IS Employee] {
+        name
+    };
+
+Backward link traversal requires to specify the original link's source
+type, but other than that it works the same way as forward traversal.
+
```

Comment:
What about `Many-to-One`?

Reply:
Many-to-one is mentioned in a backward link traversal description of one-to-many.

Reply:
That isn't enough.  `Many-to-One` is actually the most used option, so it deserves a top level mention.  Perhaps even before `One-to-One`.